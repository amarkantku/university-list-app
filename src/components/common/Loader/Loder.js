import React from 'react';
import classNames from 'classnames/bind';
import styles from './Loder.css';
let clx = classNames.bind(styles);

export default function Loder() {
    return (
        <div data-testid='spinner'>
            <svg className={clx('spinner')} viewBox="0 0 50 50">
                <circle data-testid='circle' className={clx('path')} cx="25" cy="25" r="20" fill="none" strokeWidth="5"></circle>
            </svg>  
        </div>
    )
}
