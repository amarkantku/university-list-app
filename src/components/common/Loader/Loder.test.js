import React from 'react';
import Loder from './Loder';
import { render } from '@testing-library/react';



describe("<Loder />", () => {
    test ('should component render', () => {
        const { asFragment } = render(<Loder />);
        expect(asFragment()).toMatchSnapshot();
      });
      
    test('should have getByTestId="spinner"', () => {
        const { getByTestId } = render(<Loder />);
        expect(getByTestId('spinner')).toBeTruthy();
    });
    test('should have spinner class', () => {
        const { getByTestId } = render(<Loder />);
        expect(getByTestId('circle')).toBeTruthy();
    });
})