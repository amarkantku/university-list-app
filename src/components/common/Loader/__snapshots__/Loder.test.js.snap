// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loder /> should component render 1`] = `
<DocumentFragment>
  <div
    data-testid="spinner"
  >
    <svg
      class="spinner"
      viewBox="0 0 50 50"
    >
      <circle
        class="path"
        cx="25"
        cy="25"
        data-testid="circle"
        fill="none"
        r="20"
        stroke-width="5"
      />
    </svg>
  </div>
</DocumentFragment>
`;
